src/ExcelDataReader/Core/BinaryFormat/XlsBiffStream.cs

```diff
@@ -126,12 +126,8 @@ public XlsBiffRecord Read()
 
             if (record != null)
             {
-                // Set readOption to loose to not cause exception here (sql reporting services)
-                if (_workbook.ReadOption == ReadOption.Strict)
-                {
-                    if (record.Bytes.Length < Position + record.Size)
-                        throw new ArgumentException(Errors.ErrorBiffBufferSize);
-                }
+                if (record.Bytes.Length < Position + record.Size)
+                    throw new ArgumentException(Errors.ErrorBiffBufferSize);
```

Comment:
Yes, I agree to remove that too. The next code block will return null in that case, which is a better way to signify EOF rather than blowing up.